staging/src/k8s.io/apiserver/pkg/server/filters/maxinflight.go

```diff
@@ -108,14 +102,29 @@ func startWatermarkMaintenance(watermark *requestWatermark, stopCh <-chan struct
 
 		metrics.UpdateInflightRequestMetrics(watermark.phase, readOnlyWatermark, mutatingWatermark)
 	}, inflightUsageMetricUpdatePeriod, stopCh)
+}
 
-	// Periodically observe the watermarks. This is done to ensure that they do not fall too far behind. When they do
-	// fall too far behind, then there is a long delay in responding to the next request received while the observer
-	// catches back up.
-	go wait.Until(func() {
-		watermark.readOnlyObserver.Add(0)
-		watermark.mutatingObserver.Add(0)
-	}, observationMaintenancePeriod, stopCh)
+var initMaxinflightOnce sync.Once
+
+func initMaxinflight(nonMutatingLimit, mutatingLimit int) {
```

Comment:
nit: for consistency with other places in this file: `initMaxInFlight` ?

Reply:
done